### PR TITLE
Xor and match

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: src/constant_evaluator/mod.rs [hunk 1]
```diff
@@ -91,6 +91,15 @@ impl<'a> Evaluator<'a> {
...
                 values[abstract_to_degree(&arg_values[0]) as usize % values.len()].clone()
             }
+            Expression::MatchExpression(scrutinee, arms) => {
+                let v = self.evaluate(scrutinee);
+                for (n, e) in arms {
+                    if n.is_none() || Some(&v) == n.as_ref() {
+                        return self.evaluate(e);
+                    }
+                }
+                panic!("No arm matched the value {v}");
```
Comment (Schaeff) [inline src/constant_evaluator/mod.rs:101]:
```suggestion
                arms.iter()
                    .find(|(n, _)| !matches!(n, Some(n) if *n != v))
                    .map(|(_, e)| self.evaluate(e))
                    .expect("No arm matched the value {v}")
```

**Review by Schaeff — COMMENTED** [review]
Nice!

File: src/constant_evaluator/mod.rs (hunk 1 above)
Comment (chriseth) [reply src/constant_evaluator/mod.rs:101]: I find the negated matches macro quite hard to read to be honest.

Comment (chriseth) [reply src/constant_evaluator/mod.rs:101]: Maybe `find (|(n, _)| n == None || n.as_ref() == Some(&v))`?

**Pull request merged**